BigQuery/tests/Unit/DatasetTest.php

```diff
@@ -217,4 +219,51 @@ public function testGetsIdentity()
         $this->assertEquals($this->datasetId, $dataset->identity()['datasetId']);
         $this->assertEquals($this->projectId, $dataset->identity()['projectId']);
     }
+
+    public function testsGetsModel()
+    {
+        $dataset = $this->getDataset($this->connection);
+        $model = $dataset->model($this->modelId);
+        $this->assertInstanceOf(Model::class, $model);
+        $this->assertEquals($this->datasetId, $model->identity()['datasetId']);
+        $this->assertEquals($this->projectId, $model->identity()['projectId']);
+        $this->assertEquals($this->modelId, $model->id());
+    }
+
+    public function testsGetsModelsWithoutToken()
+    {
+        $this->connection->listModels(Argument::any())
+            ->willReturn([
+                'models' => [
+                    ['modelReference' => ['modelId' => $this->modelId]]
+                ]
+            ])
+            ->shouldBeCalledTimes(1);
+
+        $dataset = $this->getDataset($this->connection);
+        $models = iterator_to_array($dataset->models());
```

Comment:
Can you assert `instanceof ItemIterator` as well?

Reply:
Sure can!